### PR TITLE
Task #210012 and Task #210087 - Added condition for single string value while navigating to assessment + Back button rendering

### PR DESCRIPTION
**Pull request by Aar-if**
### I have ensured that following `Pull Request Checklist` is taken care of before sending this PR
* [x] Code is formatted as per format decided
* [x] Updated acceptance criteria in tracker
* [x] Updated test cases in test-cases-tracker

### CODE REVIEW

File: apps/front-end/src/pages/front-end/Dashboard.js [hunk 1]
```diff
@@ -112,6 +113,48 @@ export default function Dashboard({ userTokenInfo, footerLinks }) {
...
+  //   crypto.getRandomValues(array);
+  //   const randomizedDoId = doIdArray[array[0] % doIdArray.length];
+  //   setRandom(randomizedDoId);
+  // };
+
+  const handleRandomise = async () => {
+    const doIdArray = modalVisible?.params?.do_id;
+
+    if (!doIdArray || (Array.isArray(doIdArray) && doIdArray.length === 0)) {
+      alert("There is no assessment available for this event");
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/Dashboard.js:134]: Remove javascript alert and show in modal.

File: apps/front-end/src/pages/front-end/Dashboard.js [hunk 2]
```diff
@@ -112,6 +113,48 @@ export default function Dashboard({ userTokenInfo, footerLinks }) {
...
+    crypto.getRandomValues(array);
+    const randomizedDoId = doIdArray[array[0] % doIdArray.length];
+    setRandom(randomizedDoId);
+    return randomizedDoId;
+  };
+
+  const startTest = async () => {
+    try {
+      const randomizedDoId = await handleRandomise();
+      navigate(`/assessment/events/${modalVisible.id}/${randomizedDoId}`);
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/Dashboard.js:152]: ``
navigate(`/assessment/events/${modalVisible.id}/${randomizedDoId}`);
``
change to
``
 navigate(`/assessment/events/${modalVisible?.id}/${randomizedDoId}`);
``

File: apps/front-end/src/pages/front-end/Dashboard.js [hunk 3]
```diff
@@ -112,6 +113,48 @@ export default function Dashboard({ userTokenInfo, footerLinks }) {
...
+  //   if (!doIdArray || doIdArray.length === 0) {
+  //     alert("There is no assessment availabe for this event");
+  //     return;
+  //   }
+
+  //   const array = new Uint32Array(1);
+  //   crypto.getRandomValues(array);
+  //   const randomizedDoId = doIdArray[array[0] % doIdArray.length];
+  //   setRandom(randomizedDoId);
+  // };
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/Dashboard.js:128]: Remove commented code.

**Review by sagarkoshti1990 — CHANGES_REQUESTED** [review]

File: apps/front-end/src/pages/front-end/Dashboard.js (hunk 3 above)
Comment (Aar-if) [reply apps/front-end/src/pages/front-end/Dashboard.js:128]: Okay

File: apps/front-end/src/pages/front-end/Dashboard.js (hunk 2 above)
Comment (Aar-if) [reply apps/front-end/src/pages/front-end/Dashboard.js:152]: Okay

File: apps/front-end/src/pages/front-end/Dashboard.js (hunk 1 above)
Comment (Aar-if) [reply apps/front-end/src/pages/front-end/Dashboard.js:134]: Okay

File: apps/front-end/src/pages/front-end/Dashboard.js (hunk 2 above)
Comment (Aar-if) [reply apps/front-end/src/pages/front-end/Dashboard.js:152]: Done

File: apps/front-end/src/pages/front-end/Dashboard.js (hunk 3 above)
Comment (Aar-if) [reply apps/front-end/src/pages/front-end/Dashboard.js:128]: Done

**Pull request merged**